src/OrchardCore.Modules/OrchardCore.Facebook/Widgets/Handlers/FacebookPluginPartHandler.cs

```diff
@@ -21,10 +25,28 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context,
         {
             return context.ForAsync<BodyAspect>(async bodyAspect =>
             {
+                if (_bodyAspect != null)
+                {
+                    bodyAspect.Body = _bodyAspect;
+                    return;
+                }
+
                 try
                 {
-                    var result = await _liquidTemplateManager.RenderAsync(part.Liquid, System.Text.Encodings.Web.HtmlEncoder.Default, new TemplateContext());
-                    bodyAspect.Body = new HtmlString(result);
+                    var model = new FacebookPluginPartViewModel()
+                    {
+                        Liquid = part.Liquid,
+                        FacebookPluginPart = part,
+                        ContentItem = part.ContentItem
+                    };
+
+                    var templateContext = new TemplateContext();
+                    templateContext.SetValue("ContentItem", part.ContentItem);
+                    templateContext.MemberAccessStrategy.Register<FacebookPluginPartViewModel>();
+                    templateContext.LocalScope.SetValue("Model", model);
```

Comment:
Don't use `LocalScope`

Reply:
For infos this is what we do in the `LiquidViewTemplate.ContextualizeAsync()` when a model is provided by the `displayContext.Value`. And already done in some other places.

Do i still need to change it?

Reply:
Just saw that `templateContext.SetValue("Model", model);` does exactly the same, i will replace it everywhere

Reply:
Done everywhere